testnet/tools/nns-tools/list-new-commits.sh

```diff
@@ -19,39 +19,86 @@ Usage: $0 (<COMMIT_ID>)
     exit 1
 }
 
-if [ $# -gt 1 ]; then
+RELEASE_CANDIDATE_COMMIT_ID="${1:-GARBAGE}"
+
+if [[ "${RELEASE_CANDIDATE_COMMIT_ID}" == 'GARBAGE' ]]; then
+    LATEST_COMMIT_WITH_PREBUILT_ARTIFACTS=$(latest_commit_with_prebuilt_artifacts 2>/dev/null)
+    echo
+    print_yellow "The latest commit with prebuilt artifacts: ${LATEST_COMMIT_WITH_PREBUILT_ARTIFACTS}"
     help
 fi
 
-LATEST_ARTIFACTS_COMMIT=$(latest_commit_with_prebuilt_artifacts 2>/dev/null)
+NETWORK=ic
+
+list_new_canister_commits() {
+    CANISTER_NAME="${1}"
+    CODE_DIRECTORIES="${2}"
+    LATEST_RELEASED_COMMIT_ID="${3}"
 
-RELEASE_CANDIDATE_COMMIT_ID=${1:-$LATEST_ARTIFACTS_COMMIT}
-echo "Listing commits from:" "$RELEASE_CANDIDATE_COMMIT_ID"
+    RANGE="${LATEST_RELEASED_COMMIT_ID}..${RELEASE_CANDIDATE_COMMIT_ID}"
+    NEW_COMMITS=$(
+        git \
+            --no-pager \
+            log \
+            --format="%C(auto) %h %s" \
+            "${RANGE}" \
+            -- \
+            "${CODE_DIRECTORIES}"
```

Comment:
I remember not quoting was the right thing to do because some has multiple directories.  When you quote, they come out as a single value, and weren't parsed correctly.

Reply:
Done.

Bash is a fractal of brain damage.